source/app/controllers/urls_controller.rb

```diff
@@ -1,2 +1,47 @@
 class UrlsController < ApplicationController
+  # GET /urls
+  # GET /urls.json
+  def index
+    redirect_to login_path and return if !logged_in?
+    @urls = Url.all
+  end
+
+  # GET /urls/1
+  # GET /urls/1.json
+  def show
+    url = Url.find_by(short_url: params[:id])
+    url.click_count += 1
+    url.save
+    redirect_to url.real_url
+  end
+
+  # GET /urls/new
+  def new
+    redirect_to login_path and return if !logged_in?
```

Comment:
This line is repeated a few times. Check out `before_action` to de-duplicate it.